### PR TITLE
fix: updating GitHub token to be stored as a secret

### PR DESCRIPTION
**Pull request by gausnes**
## what
* updating GitHub token to be stored as a secret, it's currently being stored as plaintext

## why
* reducing attack surface area for leaking provisioned tokens

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -382,6 +382,7 @@ resource "aws_codebuild_project" "default" {
       content {
         name  = "GITHUB_TOKEN"
         value = var.github_token
+        type  = "SECRETS_MANAGER"
```
Comment (nitrocode) [inline main.tf:385]: This would be better as an input variable.

Comment (gausnes) [reply main.tf:385]: Any rational for allowing developer to not store this token securely?

Comment (nitrocode) [reply main.tf:385]: `PARAMETER_STORE` can also be used to store secrets securely.

File: variables.tf [hunk 2]
```diff
@@ -86,6 +86,12 @@ variable "github_token" {
   description = "(Optional) GitHub auth token environment variable (`GITHUB_TOKEN`)"
 }
 
+variable "github_token_type" {
+  type        = string
+  default     = "SECRETS_MANAGER"
```
Comment (nitrocode) [inline variables.tf:91]: At Cloud Posse, we've used ssm param store as our default. I think we should be consistent here and keep it the same. If someone wants to override it then they can use this input vadiable to set it to use ssm secrets manager.

```suggestion
  default     = "PARAMETER_STORE"
```

**Review by nitrocode — APPROVED** [review]

**Comment by nitrocode** [comment]
This has been released as https://github.com/cloudposse/terraform-aws-codebuild/releases/tag/0.37.0

Thanks for the contribution!

**Pull request merged**